lib/rpmsg/rpmsg_virtio.c

```diff
@@ -347,6 +347,13 @@ static void *rpmsg_virtio_get_tx_payload_buffer(struct rpmsg_device *rdev,
 		metal_mutex_release(&rdev->lock);
 		if (rp_hdr || !tick_count)
 			break;
+
+		status = rpmsg_virtio_notify_wait(rvdev, rvdev->rvq);
```

Comment:
create a more generic function that implement also the metal_sleep_usec(RPMSG_TICKS_PER_INTERVAL);
something such as:

```
static inline int
rpmsg_virtio_wait_for_buffer(struct rpmsg_virtio_device *rvdev,
			 struct virtqueue *vq, int timeout)
{
	if (rvdev->vdev->func->notify_wait) {
			return rvdev->vdev->func->notify_wait(rvdev->vdev, vq, timeout) :
	} else {
			metal_sleep_usec(timeout);
			return RPMSG_SUCCESS;
	}
}
```

Reply:
we need pass param timeout to rpmsg_virtio_wait_for_buffer, and need `tick_count--`.
So I change the caller.